### PR TITLE
Feature: Search input

### PR DESCRIPTION
**Pull request by thomasdigby**
Keen to get some thoughts on creating a search input for the design system, it'll only be an icon and a search input type but I think for consistency it'll be worth including.

Screenshot 2021-07-15 at 17 46 28

### CODE REVIEW

**Comment by alexandrunemes** [comment]
I'm wondering if we should have some docs or at least a test that might show how to use the input in a search context. Like getting its value, doing something when the value changes etc. Or do we just leave it as unopinionated as it currently is.

**Comment by thomasdigby** [comment]
> I'm wondering if we should have some docs or at least a test that might show how to use the input in a search context. Like getting its value, doing something when the value changes etc. Or do we just leave it as unopinionated as it currently is.

Yeah definitely, this will go back to the design team to document fully the use-cases outside of the technical perspective.

I was unsure about whether this could just be an addition to the input to keep it simple but that felt like it would go against our component composition model, then this became more about consistency to ensure that all of our search inputs are the same. So with that in mind, I definitely think it's worth keeping it unopinoinated but there's no harm adding additional docs to show it in combination with a filter method or something similar.

**Comment by andoulla** [comment]
In the designs, the alignment of the icon tends to be on the right. Will it be ok to move it to the left? I like the idea of only having one alignment option to enforce consistency

File: src/components/search-input/SearchInput.mdx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: Search Input
+component: SearchInput
+description: An input used for searching
+category: Forms
+---
+
```
Comment (andoulla) [inline src/components/search-input/SearchInput.mdx:7]: It feels a bit bare: 
```suggestion

Renders the Search icon along with the input component
```

**Comment by thomasdigby** [comment]
> In the designs, the alignment of the icon tends to be on the right. Will it be ok to move it to the left? I like the idea of only having one alignment option to enforce consistency

Yeah that's an interesting point, I've generally seen it positioned to the left in most cases on the web. The reason for that is that I think it makes a distinction between the icon representing a potential clickable action or just an indicator of the type of input.

It's easy enough to make a non-breaking change with this in future so let's merge it in and discuss in more detail when we have the time.

**Pull request merged**